.cci.jenkinsfile

```diff
@@ -33,6 +33,7 @@ pod(image: 'registry.fedoraproject.org/fedora:33', runAsUser: 0, kvm: true, memo
             try {
                 cosa_cmd("kola --basic-qemu-scenarios")
                 cosa_cmd("kola run --parallel 8")
+                cosa_cmd("kola run --qemu-firmware uefi coreos.boot-mirror*")
```

Comment:
ah, instead of removing the drive (remember, this is the host part) you have to remove the PCI device (the guest part).
try `/machine/peripheral-anon/device[5]`

Reply:
This will work for `BIOS` but not for `UEFI`

Reply:
This line: https://github.com/coreos/coreos-assembler/pull/1880/files#diff-16a92b89e4fdab63ff64b808b2495d53843ca8e5b6614bb5493be3077c8726ccR348 takes care of removing the last part i.e `/virtio-backend`

Reply:
Hmm, I wonder if it's dependent on which version of QEMU the host machine is running or if it's maybe that UEFI wants a different way of specifying the device to `drive_del`, note in my test I used the nickname `d2` I got from `info block`.

QEMU version:
```
$ qemu-system-x86_64 --version
QEMU emulator version 4.2.1 (qemu-4.2.1-1.fc32)
Copyright (c) 2003-2019 Fabrice Bellard and the QEMU Project developers
```

qemuexec call:
```
kola qemuexec -D 5G -m 16384 -i mirror.ign -c --qemu-firmware uefi
```

Ignition config generated from:
```
variant: fcos
version: 1.3.0
boot_device:
  mirror:
    devices:
      - /dev/vda
      - /dev/vdb
```

First boot qemu monitor calls:
```
(qemu) info block
d1 (#block167): /dev/fdset/1 (qcow2)
    Attached to:      /machine/peripheral-anon/device[2]/virtio-backend
    Cache mode:       writeback, ignore flushes
pflash0 (#block327): /usr/share/edk2/ovmf/OVMF_CODE.fd (raw, read-only)
    Attached to:      /machine/system.flash0
    Cache mode:       writeback
pflash1 (#block536): /dev/fdset/2 (raw)
    Attached to:      /machine/system.flash1
    Cache mode:       writeback
d2 (#block768): /dev/fdset/3 (qcow2)
    Attached to:      /machine/peripheral-anon/device[3]/virtio-backend
    Cache mode:       writeback, ignore flushes
    Backing file:     /home/stephenlowrie/work/srv-rhcos/builds/47.83.202101151819-0/x86_64/rhcos-47.83.202101151819-0-qemu.x86_64.qcow2 (chain depth: 1)
ide2-cd0: [not inserted]
    Attached to:      /machine/unattached/device[34]
    Removable device: not locked, tray closed
sd0: [not inserted]
    Removable device: not locked, tray closed
(qemu) drive_del d2
(qemu) qom-s[   58.322520] blk_update_request: I/O error, dev vdb, sector 1050608 op 0x1:(WRITE) flags 0x800 phys_seg 1 prio class 0
[   58.324364] md: super_written gets error=10
[   58.325106] md/raid1:md127: Disk failure on vdb3, disabling device.
[   58.325106] md/raid1:md127: Operation continuing on 1 devices.
[   58.327545] blk_update_request: I/O error, dev vdb, sector 1050632 op 0x1:(WRITE) flags 0x800 phys_seg 1 prio class 0
[   58.329406] md: super_written gets error=10
[   58.330164] md/raid1:md126: Disk failure on vdb4, disabling device.
[   58.330164] md/raid1:md126: Operation continuing on 1 devices.
qom-set /machine/peripheral-anon/device[3]/virtio-backend bootindex -1
(qemu) qom-set /machine/peripheral-anon/device[2]/virtio-backend bootindex 1
```

Post `system_reset` output of `info block`:
```
(qemu) info block
d1 (#block167): /dev/fdset/1 (qcow2)
    Attached to:      /machine/peripheral-anon/device[2]/virtio-backend
    Cache mode:       writeback, ignore flushes
pflash0 (#block327): /usr/share/edk2/ovmf/OVMF_CODE.fd (raw, read-only)
    Attached to:      /machine/system.flash0
    Cache mode:       writeback
pflash1 (#block536): /dev/fdset/2 (raw)
    Attached to:      /machine/system.flash1
    Cache mode:       writeback
/machine/peripheral-anon/device[3]/virtio-backend: [not inserted]
    Attached to:      /machine/peripheral-anon/device[3]/virtio-backend
ide2-cd0: [not inserted]
    Attached to:      /machine/unattached/device[34]
    Removable device: not locked, tray closed
sd0: [not inserted]
    Removable device: not locked, tray closed
```

Raid device details post reboot:
```
[core@cosa-devsh ~]$ sudo mdadm --export --detail /dev/md/md-root
MD_LEVEL=raid1
MD_DEVICES=2
MD_METADATA=1.2
MD_UUID=9caaf79e:29e3b1cd:026fe81d:2c157701
MD_DEVNAME=md-root
MD_NAME=any:md-root
MD_DEVICE_dev_vda4_ROLE=0
MD_DEVICE_dev_vda4_DEV=/dev/vda4
```

Reply:
> Hmm, I wonder if it's dependent on which version of QEMU the host machine is running or if it's maybe that UEFI wants a different way of specifying the device to drive_del, note in my test I used the nickname d2 I got from info block.

For me, the `QEMU` version is: 
```
$ qemu-kvm --version
QEMU emulator version 5.1.0 (qemu-5.1.0-9.fc33)
Copyright (c) 2003-2020 Fabrice Bellard and the QEMU Project developers
```
If I use the `drive_del` command and try to run the kola test, I'm hitting the following issue:
```
failed to delete the first boot disk: Could not delete primary device d3: Running QMP command: The command drive_del has not been found
```
However, I was able to replicate the steps mentioned in your comment manually.